firewood/src/db.rs

```diff
@@ -799,15 +796,15 @@ mod test {
 
             // check the database for consistency, sometimes checking the hashes
             let hash_check = rng.borrow_mut().random();
-            if let Err(e) = db
+            let report = db
                 .check(CheckOpt {
                     hash_check,
                     progress_bar: None,
                 })
-                .await
-            {
+                .await;
+            if !report.errors.is_empty() {
                 db.dump(&mut std::io::stdout()).await.unwrap();
```

Comment:
I doubt we still want or need this. There is an option in fwdctl now that can do this.

Reply:
Oh regarding `dump`? The reason we had it here in the first place was not because of dump but because we want to fuzz testing and fuzz testing in `fwdctl` is REALLY SLOW. I will skip this comment in this PR. If you have a different opinion do let me know and I can address it in a separate PR.